src/hotspot/share/gc/z/zPage.cpp

```diff
@@ -43,7 +44,7 @@ ZPage::ZPage(ZPageType type, ZPageAge age, const ZVirtualMemory& vmem, ZMultiPar
     _multi_partition_tracker(multi_partition_tracker) {
   assert(!_virtual.is_null(), "Should not be null");
   assert((_type == ZPageType::small && size() == ZPageSizeSmall) ||
-         (_type == ZPageType::medium && size() == ZPageSizeMedium) ||
+         (_type == ZPageType::medium && size() <= ZPageSizeMediumMax && size() >= ZPageSizeMediumMin) ||
```

Comment:
Could we flip the order:
```suggestion
         (_type == ZPageType::medium && size() >= ZPageSizeMediumMin && size() <= ZPageSizeMediumMax) ||
```
Or, alternatively move the two range limits to the "outside" of the expression:
```suggestion
         (_type == ZPageType::medium && ZPageSizeMediumMin <= size() && size() <= ZPageSizeMediumMax) ||
```

or add a helper to check if a size is within the medium page range.